### ISSUE TITLE
illegal hardware instruction (core dumped) onnx convert to rknn

### ISSUE THREAD

**Issue by zhjygit** [airockchip/rknn-toolkit2#79]
[Skip to content](https://github.com/rockchip-linux/rknn-toolkit2/issues/321?tab=readme-ov-file#start-of-content)
Navigation Menu
https://github.com/rockchip-linux
/
[rknn-toolkit2](https://github.com/rockchip-linux/rknn-toolkit2)

Type / to search

[Code](https://github.com/rockchip-linux/rknn-toolkit2)
Issues
258
Pull requests
5
[Actions](https://github.com/rockchip-linux/rknn-toolkit2/actions)
[Projects](https://github.com/rockchip-linux/rknn-toolkit2/projects)
[Security](https://github.com/rockchip-linux/rknn-toolkit2/security)
[Insights](https://github.com/rockchip-linux/rknn-toolkit2/pulse)
ONNX转RKNN模型时出现了非法指令内核已转储（Illegal instruction - core dumped） #321
Open
[zhjygit](https://github.com/zhjygit) opened this issue 5 minutes ago · 0 comments
Comments
@[zhjygit](https://github.com/zhjygit)
zhjygit commented [5 minutes ago](https://github.com/rockchip-linux/rknn-toolkit2/issues/321?tab=readme-ov-file#issue-2385009838)
在j4125上部署了rknn-toolkit2，使用的是1.6版本（最新的2.0版本同样报错）
(rknpu3.9) ➜ model python ../python/convert.py ./yolov8n_opset19.onnx rk3588
W init: rknn-toolkit2 version: 1.6.0+81f21f4d
--> Config model
done
--> Loading model
Loading : 100%|████████████████████████████████████████████████| 149/149 [00:00<00:00, 12744.22it/s]
done
--> Building model
W build: found outlier value, this may affect quantization accuracy
const name abs_mean abs_std outlier value
model.0.conv.weight 2.44 2.47 -17.494
model.22.cv3.2.1.conv.weight 0.09 0.14 -10.215
model.22.cv3.1.1.conv.weight 0.12 0.19 13.361, 13.317
model.22.cv3.0.1.conv.weight 0.18 0.20 -11.216
[1] 31398 illegal hardware instruction (core dumped) python ../python/convert.py ./yolov8n_opset19.onnx rk3588
(rknpu3.9) ➜ model

(rknpu3.9) ➜ model pip list
Package Version

absl-py 2.1.0
astunparse 1.6.3
cachetools 5.3.3
certifi 2024.6.2
charset-normalizer 3.3.2
coloredlogs 15.0.1
fast-histogram 0.14
flatbuffers 24.3.25
gast 0.6.0
google-auth 2.31.0
google-auth-oauthlib 0.4.6
google-pasta 0.2.0
grpcio 1.64.1
h5py 3.11.0
humanfriendly 10.0
idna 3.7
importlib_metadata 8.0.0
keras 2.8.0
Keras-Preprocessing 1.1.2
libclang 18.1.1
Markdown 3.6
MarkupSafe 2.1.5
mpmath 1.3.0
numpy 1.23.5
oauthlib 3.2.2
onnx 1.14.1
onnxoptimizer 0.2.7
onnxruntime 1.16.0
opencv-python 4.10.0.84
opt-einsum 3.3.0
packaging 24.1
pip 24.0
protobuf 3.20.3
psutil 6.0.0
pyasn1 0.6.0
pyasn1_modules 0.4.0
requests 2.32.3
requests-oauthlib 2.0.0
rknn-toolkit2 1.6.0+81f21f4d
rsa 4.9
ruamel.yaml 0.18.6
ruamel.yaml.clib 0.2.8
scipy 1.13.1
setuptools 69.5.1
six 1.16.0
sympy 1.12.1
tensorboard 2.8.0
tensorboard-data-server 0.6.1
tensorboard-plugin-wit 1.8.1
tensorflow 2.8.0
tensorflow-io-gcs-filesystem 0.37.1
termcolor 2.4.0
tf-estimator-nightly 2.8.0.dev2021122109
torch 1.10.1
tqdm 4.66.4
typing_extensions 4.12.2
urllib3 2.2.2
Werkzeug 3.0.3
wheel 0.43.0
wrapt 1.16.0
zipp 3.19.2

@zhjygit

Add a comment
Comment
 
Add your comment here...
 
Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
Assignees
No one assigned
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
No branches or pull requests

Notifications
Customize
You’re receiving notifications because you authored the thread.
1 participant
@zhjygit
Footer
© 2024 GitHub, Inc.
Footer navigation
[Terms](https://docs.github.com/site-policy/github-terms/github-terms-of-service)
[Privacy](https://docs.github.com/site-policy/privacy-policies/github-privacy-statement)
[Security](https://github.com/security)
[Status](https://www.githubstatus.com/)
[Docs](https://docs.github.com/)
[Contact](https://support.github.com/?tags=dotcom-footer)
Manage cookies
Do not share my personal information
airockchip/rknn-toolkit2

**Comment by skoroneos** [comment]
I have been banging my head on this for a whole day.
You are missing the AVX command. from your CPU, either because you machine does not have it (old hardware) or if you are using  a VM its not passing it through.
Do a cat /proc/cpuinfo | grep avx to verify it

**Comment by yunlianwangluo** [comment]
> I have been banging my head on this for a whole day. You are missing the AVX command. from your CPU, either because you machine does not have it (old hardware) or if you are using a VM its not passing it through. Do a cat /proc/cpuinfo | grep avx to verify it

What if get null after running  cat /proc/cpuinfo | grep avx ? I am using virtualbox on AMD R7 cpu
